securedrop_client/gui/widgets.py

```diff
@@ -1006,19 +1006,22 @@ def get_selected_source(self) -> Optional[Source]:
             return source_widget.source
         return None  # pragma: nocover
 
-    def get_source_widget(self, source_uuid: str) -> Optional[QListWidget]:
+    def get_source_widget(self, source_uuid: str) -> Optional["SourceWidget"]:
```

Comment:
That mypy accepts a string is very neat!

Reply:
Since we're enforcing Python [≥ 3.9](https://github.com/freedomofpress/securedrop-client/blob/8cd7682c5d7ac5b14ff8ea1a969bd9f348fdfbf2/README.md?plain=1#L60), is there a reason to prefer the string reference over using [`from __future__ import annotations`](https://mypy.readthedocs.io/en/stable/runtime_troubles.html?highlight=forward%20referenc#class-name-forward-references)?

Reply:
I didn't know about _future annotations_ (love the name)!

I don't see any downsides to using that, and I'm keen on making the change.

---

As a note: on the same page, I do have a strong feeling that solving circular dependencies through the use of `TYPE_CHECKING` is an anti-pattern. That's bot what we're talking about, I'm only saying that, upon quick review, I'd take some of those tools with caution.